java/spark/src/main/java/com/lancedb/lance/spark/LanceIdentifier.java

```diff
@@ -14,13 +14,89 @@
 package com.lancedb.lance.spark;
 
 import org.apache.spark.sql.connector.catalog.Identifier;
+import org.apache.spark.sql.util.CaseInsensitiveStringMap;
 
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * LanceIdentifier is a custom implementation of {@link Identifier} for Lance. It contains the
+ * dataset URI and the namespace. The key difference with IdentifierImpl is that the LanceIdentifier
+ * will have the read or write options for lance. If there are some options, the `name` field will
+ * change into this format "name#key1=value1&amp;key2=value2".
+ */
 public class LanceIdentifier implements Identifier {
-  private final String[] namespace = new String[] {"default"};
+  public static final String SEPARATOR = "#";
```

Comment:
I am not sure if `#` is a more general choice. If we look HTTP URL, it uses `?`, just like this : `https://www.abc.com/tos/bucket/setting?id=aaa&region=cn-bj&dirPrefix=yh.lance%2Fdata%2F&type=objects`. JDBC URL seems also use `?`, e.g `jdbc:mysql://127.0.0.1:3306/news?useSSL=false`